src/__swaps__/utils/gasUtils.ts

```diff
@@ -617,3 +622,36 @@ export const getBaseFeeTrendParams = (trend: number) => {
 };
 
 export const chainShouldUseDefaultTxSpeed = (chainId: ChainId) => chainId === ChainId.mainnet || chainId === ChainId.polygon;
+
+const mockedGasLimit = '21000';
+
+export const useMeteorologyReport = () => {
+  const { params } = useRoute();
+  const { currentNetwork } = (params as any) || {};
+  const chainId = getNetworkObj(currentNetwork).id;
+  const { data, isLoading } = useMeteorology({ chainId });
+  const [nativeAsset, setNativeAsset] = useState<ParsedAddressAsset | undefined>();
+  const { nativeCurrency } = useAccountSettings();
+  useEffect(() => {
+    const getNativeAsset = async () => {
+      const theNativeAsset = await ethereumUtils.getNativeAssetForNetwork(currentNetwork);
+      setNativeAsset(theNativeAsset);
+    };
+    getNativeAsset();
+  }, [currentNetwork, setNativeAsset]);
+
+  let gasFeeParamsBySpeed: GasFeeParamsBySpeed | GasFeeLegacyParamsBySpeed | any = useMemo(() => {
```

Comment:
@skylarbarrera would love to figure out if we're still using legacyparams anywhere, it's giving some ts grief to some downstream consumers